Mathlib/Algebra/MonoidAlgebra/ToDirectSum.lean

```diff
@@ -147,9 +147,6 @@ theorem toDirectSum_mul [DecidableEq ι] [AddMonoid ι] [Semiring M] (f g : AddM
   simp only [AddMonoidHom.coe_mk, ZeroHom.coe_mk, toDirectSum_single]
   erw [AddMonoidAlgebra.single_mul_single, AddMonoidHom.coe_mk, ZeroHom.coe_mk,
     AddMonoidAlgebra.toDirectSum_single]
-  simp only [AddMonoidHom.coe_comp, AddMonoidHom.coe_mul, AddMonoidHom.coe_mk, ZeroHom.coe_mk,
-    Function.comp_apply, toDirectSum_single, AddMonoidHom.id_apply, Finsupp.singleAddHom_apply,
-    AddMonoidHom.coe_mulLeft]
```

Comment:
These still improve the goal state, right? I think instead of pruning this `simp only`, we should be fixing the API so that we can eliminate the `erw`.